Framework/DataObjects/src/Histogram1D.cpp

```diff
@@ -65,5 +65,18 @@ const MantidVec &Histogram1D::dataDx() const { return m_histogram.dataDx(); }
 /// Deprecated, use dx() instead.
 const MantidVec &Histogram1D::readDx() const { return m_histogram.readDx(); }
 
+/**
+ * Makes sure the histogram has valid Y data.
+ * @param histogram A histogram to process.
+ * @throw std::invalid_argument if Y data is NULL.
+ */
+void Histogram1D::checkAndSanitizeHistogram(
+    HistogramData::Histogram &histogram) {
+  if (!histogram.sharedY()) {
```

Comment:
There was a reason the checking ended up in `Histogram1D` but I cannot remember it anymore. Anyway, I made that decision for an implementation that didn't throw but instead silently filled Y and E with zeros. Now, I've moved the throwing version to `ISpectrum` and don't see problems with it. The only downside is that unit tests need to be done in `Histogram1D` as ISpectrum is too virtual for anything.

I also added check for null E data. I thought it's not possible to have null E if you give Y to `Histogram`'s constructor, but there are other ways to construct invalid `Histogram`s.